Clojure/tests/syntax_test_clojure.clj

```diff
@@ -0,0 +1,2007 @@
+;            ^^^^ constant.numeric.clojure
+  +0/10 +10/20 +30/0
+; ^^^^^ constant.numeric.clojure
+;      ^- constant
+;       ^^^^^^ constant.numeric.clojure
+;              ^^^^^ constant.numeric.clojure
+  -0/10 -10/20 -30/0
+; ^^^^^ constant.numeric.clojure
+;      ^- constant
+;       ^^^^^^ constant.numeric.clojure
+;              ^^^^^ constant.numeric.clojure
+  1234M 1234.0M 1234.1234M
+; ^^^^^ constant.numeric.clojure
+;      ^- constant
+;       ^^^^^^^ constant.numeric.clojure
+;               ^^^^^^^^^^ constant.numeric.clojure
+  +1234M +1234.0M +1234.1234M
+; ^^^^^^ constant.numeric.clojure
+;       ^- constant
+;        ^^^^^^^^ constant.numeric.clojure
+;                 ^^^^^^^^^^^ constant.numeric.clojure
+  -1234M -1234.0M -1234.1234M
+; ^^^^^^ constant.numeric.clojure
+;       ^- constant
+;        ^^^^^^^^ constant.numeric.clojure
+;                 ^^^^^^^^^^^ constant.numeric.clojure
+  1234e10 1234E10M 1234.1234e10M 1234.1234E10M
+; ^^^^^^^ constant.numeric.clojure
+;        ^- constant
+;         ^^^^^^^^ constant.numeric.clojure
+;                  ^^^^^^^^^^^^^ constant.numeric.clojure
+;                                ^^^^^^^^^^^^^ constant.numeric.clojure
+  +1234e10 +1234E10M +1234.1234e10M +1234.1234E10M
+; ^^^^^^^^ constant.numeric.clojure
+;         ^- constant
+;          ^^^^^^^^^ constant.numeric.clojure
+;                    ^^^^^^^^^^^^^^ constant.numeric.clojure
+;                                   ^^^^^^^^^^^^^^ constant.numeric.clojure
+  -1234e10 -1234E10M -1234.1234e10M -1234.1234E10M
+; ^^^^^^^^ constant.numeric.clojure
+;         ^- constant
+;          ^^^^^^^^^ constant.numeric.clojure
+;                    ^^^^^^^^^^^^^^ constant.numeric.clojure
+;                                   ^^^^^^^^^^^^^^ constant.numeric.clojure
+  1234.1234e+10 1234.1234E+10 1234.1234e-10 1234.1234E-10
+; ^^^^^^^^^^^^^ constant.numeric.clojure
+;              ^- constant
+;               ^^^^^^^^^^^^^ constant.numeric.clojure
+;                             ^^^^^^^^^^^^^ constant.numeric.clojure
+;                                           ^^^^^^^^^^^^^ constant.numeric.clojure
+  +1234.1234e+10M +1234.1234E+10M +1234.1234e-10M +1234.1234E-10M
+; ^^^^^^^^^^^^^^^ constant.numeric.clojure
+;                ^- constant
+;                 ^^^^^^^^^^^^^^^ constant.numeric.clojure
+;                                 ^^^^^^^^^^^^^^^ constant.numeric.clojure
+;                                                 ^^^^^^^^^^^^^^^ constant.numeric.clojure
+  -1234.1234e+10M -1234.1234E+10M -1234.1234e-10M -1234.1234E-10M
+; ^^^^^^^^^^^^^^^ constant.numeric.clojure
+;                ^- constant
+;                 ^^^^^^^^^^^^^^^ constant.numeric.clojure
+;                                 ^^^^^^^^^^^^^^^ constant.numeric.clojure
+;                                                 ^^^^^^^^^^^^^^^ constant.numeric.clojure
+
+; ## Breaks
+
+  10,20,30
+; ^^ constant.numeric.clojure
+;   ^ comment.punctuation.comma.clojure
```

Comment:
`comment.punctuation` seems like a funny scope name. Are the commas really comments and not some sort of separator?

Reply:
In Clojure/EDN, comments are parsed as whitespace. They disappear from the AST after parsing, so yeah, they're comments.